.github/workflows/tiobe-tics-cron.yaml

```diff
@@ -0,0 +1,80 @@
+name: TiCS Nightly Security Report
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 10 * * *'
+
+permissions:
+  contents: read
+
+jobs:
+  TiCS:
+
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          # Latest branches
+          - { branch: main }
+
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+
+      - name: Checking out repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{matrix.branch}}
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install Python Testing Deps
+        run: |
+          set -eux -o pipefail
+
+          # tox required for running the unit tests with coverage:
+          sudo python3 -m pip install tox
+
+          # required by TICSServer:
+          sudo python3 -m pip install pylint flake8
+
+      - name: Install Go for Cobertura Coverage Converter
+        uses: actions/setup-go@v5
+        with:
+          go-version: "1.22"
+
+      - name: Run Tests With Coverage
+        run: |
+          set -eux -o pipefail
+
+          pushd $GITHUB_WORKSPACE
+          tox -e unit,coverage-xml
+
+          GENERATED_COVERAGE_XML="$GITHUB_WORKSPACE/charms/worker/k8s/coverage.xml"
+          cat "$GENERATED_COVERAGE_XML"
+
+          # TiCS expects the report to be under a "$(pwd)/.coverage" directory.
+          mkdir -p "$GITHUB_WORKSPACE/.coverage"
+          mv "$GENERATED_COVERAGE_XML" .coverage/coverage.xml
+
+      - name: Run TiCS
+        run: |
+          # NOTE(aznashwan): TiCS install script doesn't define defaults; cannot '-u'
+          set -ex -o pipefail
+
+          export TICSAUTHTOKEN=${{ secrets.TICSAUTHTOKEN }}
```

Comment:
@aznashwan who should provide this TICSAUTHTOKEN?

Reply:
@aznashwan do you have any thoughts on where this secret comes from?

Reply:
Sorry for the delayed reply, lost track of this PR.

> who should provide this TICSAUTHTOKEN?

AFAIK there is one single global token shared by Canonical as whole which was given to us by Tiobe.

I have forward you an e-mail with it FWIW.